ui/src/VoxettaRecordButton.js

```diff
@@ -13,39 +13,56 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
-import {AudioRecorder} from './AudioRecorder';
 import {LitElement, html, css} from 'lit-element';
 
+import {AudioRecorder} from './AudioRecorder';
+import {UtteranceApiService} from './UtteranceApiService';
+
+/**
+ * Button responsible for enabling the user to record and upload audio files. 
+ */
 export class VoxettaRecordButton extends LitElement {
     static get properties() {
         return {
             isRecording: {type: Boolean},
             audioRecorder: {type: Object},
+            utteranceService: {type: Object}
         };
     }
+
     constructor() {
         super();
         this.isRecording = false;
         this.audioRecorder = new AudioRecorder();
+        this.utteranceService = new UtteranceApiService(); 
     }
-    render() {  
+
+    render() {
         return html`
             <button @click=${this.recordHandler}>Record Voice</button>
             <audio id="utterance" controls src="" style="display: none"></audio>
         `;
     }
 
-     async recordHandler(e){
+    /**
+     * If the user is not currently recording, begin recording using the Microphone component. Otherwise,
```

Comment:
nit: wrap this line, comment is too long.

Reply:
adjusted